### ISSUE TITLE
Specs for AWS and Hosting

### ISSUE THREAD

**Issue by arthursoares** [envisioning/techdb#53]
- [x] Modulus (Node.js) with SearchLy AddOn (ElasticSearch)
- [x] AWS S3 (CDN)
- [x] Cloudinary (Image CDN)
- [x] Compose.io (MongoDB)
- [ ] MailGun (SMTP)

@rafaelcorreiapoli  @lnmunhoz 

Can we move forward with our hosting setup/solution for TDB? This will also be needful for Godzilla. 

What are your plans? 

* Hosting and Server on AWS? 
* DB? 
* Image CDN?

Please give me instructions so we can set up our account.

**Comment by rafaelcorreiapoli** [comment]
**Hosting**
We could go for Heroku or Modulus.io. Both are great for MeteorJS applications. Modulus.io has a better GUI for managing apps but it's more expensive.

**Elastic Search**
Elastic Search is an expensive service. Although, if we go with Modulus.io for hosting, there is an add-on for Elasticsearch provided by Searchly where we get 500mb for free, which is "a lot". If you prefer to pay for one, we could set up an Elasticsearch instance on compose.io.

**DB**
We must go with compose.io, it's the most used by meteor community because they have support for OPLOG and it's essential for MeteorJS application's performance on live queries

**File Hosting**
We could set up an instance on AWS's S3

**Image CDN**
For user profile pictures, there is a very nice service called Cloudinary. They have features for processing the picture via parameters on request and they provide a lot of cool image transformations, like face detection to generate user thumbs/avatar and things like that. Their free plan covers a lot, but if you don't find necessary, we could also host profile pictures on S3

**Comment by arthursoares** [comment]
*Hosting / Elastic Search*
Let's take Modulus, it gives the added bonus of Elastic Search. Could you advise on which plan to get? Could this be used for static / php pages as well?

*DB*
Which plan?

*Hosting*
Great. Suggestions on plan?

*CDN*
We like it. Free plan sound great.

**Comment by michellzappa** [comment]
What’s the approximate total monthly cost?

**Comment by rafaelcorreiapoli** [comment]
Modulus supports PHP, but I don't know if we could host multiple applications on the same Servo, I will try to find out. @lnmunhoz Do you know anything about this?
I suggest to start with the 28.8$ plan because of hardware specs needed for Meteor apps

For the DB, MongoDB 1GB on compose.io is fine

AWS S3 free tier is fine for the first months, I think

We can always start with the basic plans and scale as the app needs. All of those services provide easy scale methods

**Comment by michellzappa** [comment]
Cooooool. Thanks.

**Comment by rafaelcorreiapoli** [comment]
**Option 1:**
Modulus: 28.8
ElasticSearch Add-on: 500mb for free
Compose.io (MongoDB): 1GB for 18$
S3: Free for the first months
Cloudinary: Free for a looooooong time

**Option 2:**
Heroku: 7$
Searchly (ElasticSearch): $19 for 1gb
Compose.io (MongoDB): 1GB for 18$
S3: Free for the first months
Cloudinary: Free for a looooooong time

**Comment by rafaelcorreiapoli** [comment]
We also need a service for sending emails, we could go for Mailgun https://www.mailgun.com/pricing
and it will be free for a long time...

**Comment by rafaelcorreiapoli** [comment]
Approximate total: around R$ 200

**Comment by arthursoares** [comment]
Sweet. :)

**Comment by michellzappa** [comment]
Very doable. Thanks for this!

**Comment by arthursoares** [comment]
Is DigitalOcean a valid option?

**Comment by rafaelcorreiapoli** [comment]
I never worked with Meteor on Digital Ocean but I know it's possible and made easy with this https://github.com/arunoda/meteor-up
But I like the UI of Heroku and Modulus and the possibily to use Addons, they make integration with external services very easy

**Comment by arthursoares** [comment]
[cloudinary.txt](https://github.com/envisioning/techdb/files/108573/cloudinary.txt)

**Comment by arthursoares** [comment]
Criei nossa conta no Modulus.io, Cloudinary e AWS.

@rafaelcorreiapoli  @lnmunhoz  passem os infos de users de vocês de Modulus e AWS (ou e-mails)

**Comment by arthursoares** [comment]
@rafaelcorreiapoli https://gist.github.com/arthursoares/3995bcd28fcd5751b010

**Comment by arthursoares** [comment]
@rafaelcorreiapoli 

```
curl -s --user 'api:key-39b9976f4a6e7f40987e611dce5b14b7' \
    https://api.mailgun.net/v3/sandbox99adfe5f54974a3ba50a3f05161045fd.mailgun.org/messages \
    -F from='Mailgun Sandbox <postmaster@sandbox99adfe5f54974a3ba50a3f05161045fd.mailgun.org>' \
    -F to='Arthur Soares <as@envisioning.io>' \
    -F subject='Hello Arthur Soares' \
    -F text='Congratulations Arthur Soares, you just sent an email with Mailgun!  You are truly awesome!' 

You can see a record of this email in your logs: https://mailgun.com/cp/log 

You can send up to 300 emails/day from this sandbox server. Next, you should add your own domain so you can send 10,000 emails/month for free.'
```

**Comment by rafaelcorreiapoli** [comment]
@lnmunhoz Configura o MAIL_URL por favor

**Comment by lnmunhoz** [comment]
Fixed MAIL_URL and invite system e6d1266ee13c13d42c8415f7cdebe6ad3a01cea9
